File: 35-SearchInsertPosition-binarySearch.java
#!/usr/bin/java

class Solution {
    public int searchInsert(int[] nums, int target) {
        // binary search
        int low = 0, high = nums.length -1;
        while (low <= high){
            int mid = low + (high - low) / 2;
            if (target == nums[mid]){
                return mid;    
            } else if(target < nums[mid]){
                high = mid -1;    
            } else{ //if(target > nums[mid])
                low = mid + 1;
            }
        }
        return low;
    }
}

File: LeetCodePython/35-SearchInsertPosition-BinarySearch.py
#  https://leetcode.com/problems/search-insert-position/

class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        # solution: binary search

        start, end = 0, len(nums)-1
        while start <= end:
            mid = (start+end) // 2

            if nums[mid] == target:
                return mid
            elif target < nums[mid]:
                end = mid-1
            elif target > nums[mid]:
                start = mid+1
        
        return start
        
